services/w3c/w3c-validation.service.js

```diff
@@ -38,18 +45,49 @@ export default class W3cValidation extends BaseJsonService {
     queryParamSchema,
   }
 
-  static examples = [
-    {
-      title: 'W3C Validation',
-      namedParams: { parser: 'html' },
-      queryParams: {
-        targetUrl: 'https://validator.nu/',
-        preset: 'HTML, SVG 1.1, MathML 3.0',
+  static openApi = {
+    '/w3c-validation/{parser}': {
+      get: {
+        summary: 'W3C Validation',
+        description,
+        parameters: [
+          pathParam({
+            name: 'parser',
+            example: 'html',
+            schema: { type: 'string', enum: this.getEnum('parser') },
+            description: parserDescription,
+          }),
+          queryParam({
+            name: 'targetUrl',
+            example: 'https://validator.nu/',
+            required: true,
+            description: 'URL of the document to be validate',
+          }),
+          queryParam({
+            name: 'preset',
+            example: 'HTML, SVG 1.1, MathML 3.0',
+            description:
+              'This is used to determine the schema for the document to be valdiated against.',
+            schema: {
+              type: 'string',
+              enum: [
+                'HTML, SVG 1.1, MathML 3.0',
```

Comment:
I'm slightly in 2 minds about this one. I made this an enum in the frontend because the docs just listed a bunch of acceptable values so it seems like giving the user a dropdown to pick from seems like a more useful frontend. The validation isn't just a list of allowed values though. It is this monsterous soup of regex 😱 https://github.com/badges/shields/blob/7d7efaec22b644368b9803e5e03441b547b7476b/services/w3c/w3c-validation-helper.js#L1-L12
I'm tempted to just make the validation a list of allowed strings, but I'm slightly unsure how we ended up with this and whether there are really valid values that pass that validation but aren't in that list

Reply:
Maybe it's related to case sensitivity? https://github.com/badges/shields/pull/4148#discussion_r333752460

Reply:
To avoid breaking badges for folks who used a value with different casing, we could validate all values in a non-case-sensitive way.

Reply:
Thanks for the link to that thread.

I went to look at changing this and the read the tests (which I should have done before).

https://github.com/badges/shields/blob/880c1fb49cd50ca5a0558075c956c7cab2d7676e/services/w3c/w3c-validation-helper.spec.js#L16-L50

Changing it to an allow list but making it case insensitive should be do-able with `.insensitive()` but having read the tests the other thing we are currently tolerant to is variations in spacing. We wouldn't be able to handle that without some kind of pre-processing to normalize the spacing before validation so I think in order to move to an allow-list, we'd have to make a behaviour change.

Given that, I think I'm inclined to just leave this as-is We show a dropdown in the frontend but leave the existing validation in place for now.